src/Microsoft.AspNet.Hosting/Internal/HostingEngine.cs

```diff
@@ -87,6 +88,7 @@ public virtual IDisposable Start()
                 _applicationLifetime.NotifyStopping();
                 server.Dispose();
                 _applicationLifetime.NotifyStopped();
+                loggerFactory.Dispose();
```

Comment:
Consider in the test scenario where we start multiple TestHost/Servers in one process.  HostingEngine is only a scoped subset of an actual application, so I think DI scopes are an appropriate concept here.  Per-request scopes would be nested inside of the per-HostingEngine scopes.